### PR TITLE
:bug: (import) date formats supporting digits without leading zeros

### LINKED ISSUE #607: Invalid dates when importing from CSV
When importing transactions from a CSV using the DD/MM/YYYY format, a small percentage of date values are interpreted invalid.

image

### PR DESCRIPTION
**Pull request by MatissJanis**
Closes #607

Demo import file: 
[2020-12-sanitised.csv](https://github.com/actualbudget/actual/files/10828275/2020-12-sanitised.csv)

Before: half the dates get reported as invalid

After: import works as expected

### CODE REVIEW

File: packages/loot-design/src/components/modals/ImportTransactions.test.js [hunk 1]
```diff
@@ -2,122 +2,140 @@ import { parseDate } from './ImportTransactions';
 
 describe('Import transactions', function () {
   describe('date parsing', function () {
```
Comment (MatissJanis) [inline packages/loot-design/src/components/modals/ImportTransactions.test.js:4]: Suggestion when reviewing:
Screenshot 2023-02-24 at 19 45 45

File: packages/loot-design/src/components/modals/ImportTransactions.test.js [hunk 2]
```diff
@@ -2,122 +2,140 @@ import { parseDate } from './ImportTransactions';
...
-            '24/12/20 ',
-            '24 12-20 ',
-          ],
-        },
-      ];
+    const validInputs = [
+      {
+        order: 'yyyy mm dd',
+        cases: [
+          ['20201224', '2020-12-24'],
```
Comment (MatissJanis) [inline packages/loot-design/src/components/modals/ImportTransactions.test.js:43]: All the test cases are the same as before with a few new ones.

File: packages/loot-design/src/components/modals/ImportTransactions.js [hunk 3]
```diff
@@ -43,6 +43,14 @@ export function parseDate(str, order) {
...
   const dateGroups = (a, b) => str => {
+    const parts = str
+      .replace(/^[^\d]+/, '')
+      .replace(/[^\d]+$/, '')
+      .split(/[^\d]+/);
+    if (parts.length >= 3) {
+      return parts.slice(0, 3);
+    }
+
     const digits = str.replace(/[^\d]/g, '');
```
Comment (MatissJanis) [inline packages/loot-design/src/components/modals/ImportTransactions.js:54]: Keeping the old logic for backwards compatability. The new logic allows all sorts of weird formats, but it does not allow this one: `20230224` (this is todays date all concatenated in a single string)

**Comment by j-f1** [comment]
Suggestion: add tests for full ISO8601 dates like `2023-01-19T02:36:52` (which currently can’t be parsed but I think this PR will correctly handle).

**Comment by MatissJanis** [comment]
Good idea @j-f1 

https://github.com/actualbudget/actual/pull/692/commits/48518bf5f799f7b4a366eed47c5ee90d00fdb171

**Review by j-f1 — APPROVED** [review]

**Pull request merged**